configure.php

```diff
@@ -0,0 +1,528 @@
+			[ $arg, $value ] = explode( '=', substr( $value, 2 ), 2 );
+
+			$args[ $arg ] = trim( $value );
+
+			$previous_key = null;
+		} else {
+			$args[ substr( $value, 2 ) ] = true;
+
+			$previous_key = substr( $value, 2 );
+		}
+	} elseif ( ! empty( $previous_key ) ) {
+		$args[ $previous_key ] = trim( $value );
+	} else {
+		$previous_key = trim( $value );
+	}
+}
+
+$terminal_width = (int) exec( 'tput cols' );
+
+function write( string $text ): void {
+	global $terminal_width;
+	echo wordwrap( $text, $terminal_width - 1 ) . PHP_EOL;
+}
+
+function ask( string $question, string $default = '', bool $allow_empty = true ): string {
+	while ( true ) {
+		write( $question . ( $default ? " [{$default}]" : '' ) . ': ' );
+		$answer = readline( '> ' );
+
+		$value = $answer ?: $default;
+
+		if ( ! $allow_empty && empty( $value ) ) {
+			echo "This value can't be empty." . PHP_EOL;
+			continue;
+		}
+
+		return $value;
+	}
+}
+
+function confirm( string $question, bool $default = false ): bool {
+	write( "{$question} (yes/no) [" . ( $default ? 'yes' : 'no' ) . ']: ' );
+
+	$answer = readline( '> ' );
+
+	if ( ! $answer ) {
+		return $default;
+	}
+
+	return in_array( strtolower( trim( $answer ) ), [ 'y', 'yes', 'true', '1' ], true );
+}
+
+function run( string $command, string $dir = null ): string {
+	$command = $dir ? "cd {$dir} && {$command}" : $command;
+
+	return trim( (string) shell_exec( $command ) );
+}
+
+function str_after( string $subject, string $search ): string {
+	$pos = strrpos( $subject, $search );
+
+	if ( $pos === false ) {
+		return $subject;
+	}
+
+	return substr( $subject, $pos + strlen( $search ) );
+}
+
+function slugify( string $subject ): string {
+	return strtolower( trim( (string) preg_replace( '/[^A-Za-z0-9-]+/', '-', $subject ), '-' ) );
+}
+
+function title_case( string $subject ): string {
+	return ensure_capitalp( str_replace( ' ', '_', ucwords( str_replace( [ '-', '_' ], ' ', $subject ) ) ) );
+}
+
+function ensure_capitalp( string $text ): string {
+	return str_replace( 'Wordpress', 'WordPress', $text );
+}
+
+/**
+ * @param string $file
+ * @param array<string, string> $replacements
+ */
+function replace_in_file( string $file, array $replacements ): void {
+	$contents = file_get_contents( $file );
+
+	if ( empty( $contents ) ) {
+		return;
+	}
+
+	file_put_contents(
+		$file,
+		str_replace(
+			array_keys( $replacements ),
+			array_values( $replacements ),
+			$contents,
+		)
+	);
+}
+
+function remove_readme_paragraphs( string $file ): void {
+	$contents = file_get_contents( $file );
+
+	if ( empty( $contents ) ) {
+		return;
+	}
+
+	file_put_contents(
+		$file,
+		trim( (string) preg_replace( '/<!--delete-->.*<!--\/delete-->/s', '', $contents ) ?: $contents ),
+	);
+}
+
+function determine_separator( string $path ): string {
+	return str_replace( '/', DIRECTORY_SEPARATOR, $path );
+}
+
+/**
+ * @return array<int, string>
+ */
+function list_all_files_for_replacement(): array {
```

Comment:
Would be great to have this as a more easily configured list of files and directories that is programmatically compiled into the `grep` command. That way, if we needed to modify the list, we're just adding/removing/changing entries in two arrays (one for files and one for directories).